python/lsst/meas/extensions/multiprofit/utils.py

```diff
@@ -82,14 +85,42 @@ def get_spanned_image(
         return None, bbox
     if spans is None:
         spans = footprint.getSpans().asArray()
-    sig_inv = None
-    img = afwImage.Image(bbox, dtype="D")
-    maskedIm = exposure.photoCalib.calibrateImage(exposure.maskedImage.subset(bbox))
-    img.array[spans] = maskedIm.image.array[spans]
-    if get_sig_inv:
-        sig_inv = afwImage.Image(bbox, dtype="D").array
-        sig_inv[spans] = 1 / np.sqrt(maskedIm.variance.array[spans])
-    return img.array, bbox, sig_inv
+    sig_inv = afwImage.ImageF(bbox) if get_sig_inv else None
+    img = afwImage.ImageF(bbox)
+    img.array[:] = np.nan
+    if footprint is None:
+        maskedIm = exposure.maskedImage.subset(bbox)
+        if not calibrate:
+            img = maskedIm.image.array
+            sig_inv.array[spans] = 1 / np.sqrt(maskedIm.variance.array[spans])
+    else:
+        img.array[spans] = footprint.getImageArray()
```

Comment:
To answer your question from slack, assuming that you have correctly loaded the scarlet models and populated the catalog footprints with the scarlet models then this is correct. The while idea of the new API was to make it so that once the scarlet models were loaded, the catalogs could be processed the same regardless of which deblender created the heavy footprints.